Removed "highload" as it is un-needed

nitantsoni opened the pull request:
The script is designed to set 3 CPU frequencies - min, medium & max. Before this change, it would check if the CPU was above "highload" before setting it to max speed. This would mean that the script would be NOOP, between "midload" & "highload" values.
Changed that, so it goes to max, above midload value. 

Other scaling behaviour is unchanged.

nitantsoni committed:
Removed "highload" as it is un-needed

The script is designed to set 3 CPU frequencies - min, medium & max. Before this change, it would check if the CPU was above "highload" before setting it to max speed. This would mean that the script would be NOOP, between "midload" & "highload" values.
Changed that, so it goes to max, above midload value. 

Other scaling behaviour is unchanged.

nitantsoni committed:
Updated default load values

Set the default load values to something more sensible. 
- Low is set to 10%
- Mid is set to 20%

nitantsoni committed:
Quit if script is already running

The script will quit if it detects another process of the same name, as it's own. This is specially useful to add it as a Scheduled Task in Synology, as then it can be set to start everyday, without any drawbacks

nitantsoni commented:
Added in 2 more commits. One updates the default value to 10 & 20 of lowload & midload. The second commit is to quit if the script is running in the background already. 

This allows scheduling it to start daily via the Syno Task Scheduler

Trauma commented:
Thanks fo the MR; effectively there was a NOOP, even though it wasn't affecting in the script behavior. Less is better, so i'll merge that (i've already integrate that in the temp-guard branch).

Anyway, why changing de default load values ? I'll need to understand that before accepting your PR.

nitantsoni commented:
I feel these will lead to a more snappier system

nitantsoni committed:
Update scaler.sh

Remove script check code. Not working
Changed loadavg limits to more sensible values.

Trauma committed:
Merge branch 'master' into master

Trauma commented:
I've merged the main fix, high load noop. Thanks. (not a clean merge on my side, anyway the code is here)

I did not integrate your "load" settings, since it's overridable it do not make sense to me (but i understand you prefer other settings).

About the failsafe features :
- detect if already is a good idea, do you mind to reintegrate it ?
- I was thinking to add a condition to check if the script is running as root too, what do you think ?

nitantsoni commented:
There was a mistake. Turns out my code for the PID check is faulty and will prevent the script from starting. 

Also, while checking further on those load settings, I realized that the loadavg value is dependent on the CPU Cores. So on a 4 Core CPU like my J4105, it goes all the way upto 4.0. This makes my "load" settings a very bad idea. I will make a PR to fix this, so that the "load" values then are more intuitive (20 = 20% load)

nitantsoni commented:
I am not sure what you mean by the "failsafe" features. 

For the script check, the best way would be to use a lockfile, but that can cause issues if say the script exits uncleanly and leaves the lockfile intact. I am currently using "flock" like so 

"flock -n /tmp/cpuscale.lock /volume1/homes/nitant/cpuscale.sh"

Finally, the root check, makes a lot of sense. If it has an unclean exit on that check, Task Scheduler might even notify the user of it by email

Trauma commented:
The cleaniest way to implement that is to handle signals in the script, so
we could create/delete a PID file an so on...
https://linuxhint.com/bash_trap_command/

Le lun. 23 déc. 2019 à 17:22, nitantsoni <notifications@github.com> a
écrit :

> I am not sure what you mean by the "failsafe" features.
>
> For the script check, the best way would be to use a lockfile, but that
> can cause issues if say the script exits uncleanly and leaves the lockfile
> intact. I am currently using "flock" like so
>
> "flock -n /tmp/cpuscale.lock /volume1/homes/nitant/cpuscale.sh"
>
> Finally, the root check, makes a lot of sense. If it has an unclean exit
> on that check, Task Scheduler might even notify the user of it by email
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Trauma/cpufreq-userspace-scaler/pull/1?email_source=notifications&email_token=AAAUIS4FC2VLWL5FNQHHYFDQ2DQTJA5CNFSM4J24ODM2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHROGPI#issuecomment-568517437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAUISYBZKVUJKUU27OJ76DQ2DQTJANCNFSM4J24ODMQ>
> .
>
